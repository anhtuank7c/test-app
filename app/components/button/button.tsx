import React from 'react';
import {
  TouchableOpacity,
  Text,
  TouchableOpacityProps,
  StyleSheet,
} from 'react-native';

interface Props extends TouchableOpacityProps {
  text: String;
  color: 'red' | 'green' | 'yellow' | 'purple';
}
export const Button = ({color, style, text, ...rest}: Props) => {
  return (
    <TouchableOpacity
      style={[styles.btn, {backgroundColor: color}, style]}
      {...rest}>
      <Text style={styles.btnText}>{text}</Text>
    </TouchableOpacity>
  );
};
const styles = StyleSheet.create({
  btn: {
    backgroundColor: 'red',
    paddingVertical: 16,
    borderRadius: 10,
    marginTop: 26,
  },
  btnText: {
    fontSize: 17,
    letterSpacing: 0.2,
    alignSelf: 'center',
    color: '#fff',
  },
});
