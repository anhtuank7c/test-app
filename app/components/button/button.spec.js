import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {Button} from './button';

test('render correctly', () => {
  render(<Button />);
  expect(screen).toMatchSnapshot();
});
