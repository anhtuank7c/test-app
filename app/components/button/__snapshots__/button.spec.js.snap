// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
      "borderRadius": 10,
      "marginTop": 26,
      "opacity": 1,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "fontSize": 17,
        "letterSpacing": 0.2,
      }
    }
  />
</View>
`;
